### PR TITLE
Week6

### PR DESCRIPTION
**Pull request by shiv123-max**

### CODE REVIEW

File: Week-6/6.1/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Brute Force - Time Complexity - O(n2)
+function maxSumSubarray(arr) {
```
Comment (kapilggg10) [inline Week-6/6.1/index.js:2]: It's good to see comparison of two algorithms while solving a problem 👍

File: Week-6/6.1/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  return maxSum;
+}
+
+// Optimised Approach - Using Kadane's Algorithm - Time Complexity - O(n)
+function maxSumSubarrayKadane(arr) {
+  let maxSum = -Infinity;
+  let currSum = 0;
+  for (let i = 0; i < arr.length; i++) {
```
Comment (kapilggg10) [inline Week-6/6.1/index.js:19]: let's assume I don't know programming, can you explain me this loop in layman terms (simple language)?

File: Week-6/6.2/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  let answer = [];
+  let rowStart = 0;
+  let colStart = 0;
+  let rowEnd = row - 1;
+  let colEnd = col - 1;
+  let total = row * col;
+  let count = 0;
+
+  while (count < total) {
```
Comment (kapilggg10) [inline Week-6/6.2/index.js:13]: I have seen the same approach by multiple people, were you able to think of any other solution?
I am good with this solution, just want to know if there was any approach which you tried on your own on the first go.

File: Week-6/6.3/index.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+//One Pass Solution - Time Complexity - O(n)
+function sortZerosOnesTwosOnePass(arr) {
+  let low = 0;
+  let mid = 0;
+  let high = arr.length - 1;
+  let temp = 0;
+
+  while (mid <= high) {
```
Comment (kapilggg10) [inline Week-6/6.3/index.js:38]: can you comment before each block what is reason for using that block.
blocks: if, else, while, for etc.

File: Week-6/6.3/index.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    if (arr[mid] === 0) {
+      temp = arr[mid];
```
Comment (kapilggg10) [inline Week-6/6.3/index.js:40]: can you think of a short and sweet way of swapping two variables in js?

File: Week-6/6.5/index.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      start = mid + 1;
+    }
+    mid = Math.floor((start + end) / 2);
+  }
+
+  return false;
+}
+
+// Time Complexity - O(nlogn)
+function findPairWithDifference(arr, target) {
```
Comment (kapilggg10) [inline Week-6/6.5/index.js:19]: can you explain each of the strategy with simple language or add some comments in code so that I can be sure that you didn't blindly copied the code and understood the solution properly.

**Review by kapilggg10 — APPROVED** [review]
Try to add some comments in the code for your future reference to understand all of the approach in simpler terms.

File: Week-6/6.3/index.js (hunk 4 above)
Comment (shiv123-max) [reply Week-6/6.3/index.js:38]: Okay, I will do that.

File: Week-6/6.3/index.js (hunk 5 above)
Comment (shiv123-max) [reply Week-6/6.3/index.js:40]: Yes, I have seen it. It's like [arr[mid], arr[high]] = [arr[high], arr[mid]].

File: Week-6/6.5/index.js (hunk 6 above)
Comment (shiv123-max) [reply Week-6/6.5/index.js:19]: Okay, I will do that in the next assignments.

File: Week-6/6.2/index.js (hunk 3 above)
Comment (shiv123-max) [reply Week-6/6.2/index.js:13]: No, I was not able to think of any other solution. I got this approach in my mind from starting but I was not able to implement it successfully. Then, I took some help and implemented it.

File: Week-6/6.1/index.js (hunk 2 above)
Comment (shiv123-max) [reply Week-6/6.1/index.js:19]: Yeah, we can think of it like people sitting next to each other at a dinner table. We go to each person one by one until we reach the end of the table and do some action with that specific person. After entertaining one person, we go to the next person, and so on.
